### ISSUE TITLE
docker build task is hanging

### ISSUE THREAD

**Issue by wgorder** [Transmode/gradle-docker#37]
This is the last line of output I see 

```
12:17:59.618 [INFO] [se.transmode.gradle.plugins.docker.DockerTask] Using the native docker binary.
> Building 95% > :server:dockerBuild

```

I can run the command manually with no issues.  I am not sure what is going on.

**Comment by petehayes** [comment]
I have run into the same problem.  It is because in the NativeDockerClient, when it executes the docker command it does a waitFor() on the process. If the docker build process generates a lot of output to standard out or standard err, then it will eventually fill a buffer of the executing process and the process will block indefinitely while trying to write to the stream.  Whenever processes are executed, you have to consume the standard out and standard err or ensure that there is little to no output in there.

**Comment by cdancy** [comment]
So the solution is to NOT use this task but instead fall back to using exec? @petehayes maybe we should put together a PR to solve the issue? I'm not sure how active this project is anymore judging from lack of commits going in

**Comment by petehayes** [comment]
Yes, I was going to give that a shot at some point to fix this and issue a pull request.

**Comment by magnus-larsson** [comment]
Hi, what is the status of this issue?

From my understanding this issue is, in many cases, a blocker for using the plugin on Windows?
(never seen the problem on OS X)

See for example a discussion on the subject in the following blog post: http://callistaenterprise.se/blogg/teknik/2015/06/08/building-microservices-part-4-dockerize-your-microservices/

Regards,
Magnus.

**Comment by petehayes** [comment]
In our case, we decided to move to the other [gradle docker plugin](https://github.com/bmuschko/gradle-docker-plugin) so I won't be contributing a patch.

**Comment by magnus-larsson** [comment]
Ok, thanks for the update. I just realised that it already exists a pull request that seems to solve the issue so I'll post a comment there as well.
